datadog/dogstatsd/base.py

```diff
@@ -398,7 +399,13 @@ def _send_to_server(self, packet):
         except socket.timeout:
             # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
             pass
-        except (socket.error, socket.herror, socket.gaierror) as se:
+        except socket.error as se:
```

Comment:
I think you need to switch the two `except` statements order here because:
```
>>> isinstance(socket.herror(), socket.error)
True
```
So
```python
>>> try:
...     raise socket.herror()
... except socket.error:
...     print("1")
... except socket.herror:
...     print("2")
...
1
```

(my bad I wrote the initial suggestion in the wrong order, sorry)

Reply:
Re-ordered them in f4bec108947016266e98f901a0ab3bf5f407e132, thanks!

Reply:
Correct me if I am wrong @mrknmc but since Python 3 raises `BlockingIOError` and not `socket.error`, we'd also need to take this into consideration to be fully py2/py3 compatible?

Reply:
`BlockingIOError` raised in Python 3 is still a `socket.error` with `errno == errno.EAGAIN`:

```
>>> e = BlockingIOError(11, 'Resource temporarily unavailable')
>>> import errno
>>> e.errno == errno.EAGAIN
True
>>> import socket
>>> e.errno == errno.EAGAIN
True
```

Reply:
Are you sure about this?

https://docs.python.org/3/library/exceptions.html#BlockingIOError indicates it's an `OSError`, not a `socket.error`.

Reply:
See following docs from 3.3 release:
https://docs.python.org/3/whatsnew/3.3.html#pep-3151-reworking-the-os-and-io-exception-hierarchy

Notably:
> You don’t have to worry anymore about choosing the appropriate exception type between OSError, IOError, EnvironmentError, WindowsError, mmap.error, socket.error or select.error. All these exception types are now only one: OSError. The other names are kept as aliases for compatibility reasons.

Reply:
OK thanks! We're testing against Python 3.5 and upwards, so LGTM.

Reply:
Right, so you're actually pointing toward my direction @mrknmc. :)

We're catching `socket.error` here, not `OSError`. So `BlockingIOError` won't be caught by this code, they'll fall under the last `except Exception`.

Reply:
I think you misunderstood @jd - `socket.error` is an alias for `OSError` on Python >= 3.3. If you try the gist I linked in #514 you can convince yourself of this :)

Reply:
Alright

Reply:
Alright! I misread the last sentence and I thought they were just subclasses, not _aliasses_.

So we should be good with that for Python 2 and Python 3!